### ISSUE TITLE
Uyuni Bugreport - Uyuni 2021.12 - 2022.01 Upgrade has vendor changes again

### ISSUE THREAD

**Issue by lumarel** [uyuni-project/uyuni#4742]
# Additional Information

I already discussed this topic with Julio throughout the last year on Gitter.
There were some unwanted changes of the vendor somewhere last year, and after that it now happened on every upgrade that some packages need to change their vendor.

Interestingly everytime the Uyuni-Server-release Package, so also this time again (past reports added to the more details section):
```bash
# zypper up --allow-vendor-change
Loading repository data...
Reading installed packages...

The following 260 packages are going to be upgraded:

...

The following pattern is going to be upgraded:
  uyuni_server

The following product is going to be upgraded:
  Uyuni Server  2021.12-0 -> 2022.01-0

The following 2 packages are going to change vendor:
  python-dmidecode   SUSE LLC <https://www.suse.com/> -> obs://build.opensuse.org/systemsmanagement:Uyuni
  python3-dmidecode  SUSE LLC <https://www.suse.com/> -> obs://build.opensuse.org/systemsmanagement:Uyuni

The following product is going to change vendor:
  Uyuni Server  openSUSE -> obs://build.opensuse.org/systemsmanagement:Uyuni

...

    Note: System reboot required.
Continue? [y/n/v/...? shows all options] (y):
```

In the last discussion I was asked to add the package info from the time before the upgrade, to make sure how it looked right before the upgrade:
```bash
# rpm -q Uyuni-Server-release --info
Name        : Uyuni-Server-release
Version     : 2021.12
Release     : 160.4.uyuni1
Architecture: x86_64
Install Date: Fri Dec 10 08:29:46 2021
Group       : System/Fhs
Size        : 1455
License     : GPL-2.0-or-later
Signature   : RSA/SHA256, Mon Dec  6 22:59:35 2021, Key ID 972e5d6c0d20833e
Source RPM  : Uyuni-Server-release-2021.12-160.4.uyuni1.src.rpm
Build Date  : Mon Dec  6 22:59:28 2021
Build Host  : sheep86
Relocations : (not relocatable)
Vendor      : obs://build.opensuse.org/systemsmanagement:Uyuni
Summary     : Uyuni Server
Description :
Uyuni lets you efficiently manage physical, virtual,
and cloud-based Linux systems. It provides automated and cost-effective
configuration and software management, asset management, and system
provisioning.
Distribution: systemsmanagement:Uyuni:Master / openSUSE_Leap_15.3
```

This looks like it should already be correct and upgrade without any issue, but it doesn't wan't to without the allow-vendor-change option.

## Version of Uyuni Server after the upgrade
```bash
Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : Uyuni Server Stable
Name           : Uyuni-Server-release
Version        : 2022.01-165.2.uyuni1
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : out-of-date (version 2021.12-160.4.uyuni1 installed)
Source package : Uyuni-Server-release-2022.01-165.2.uyuni1.src
Summary        : Uyuni Server
Description    :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.
```

## Details about issue
Here are also the 2 reports from the upgrades before this one:
![image](https://user-images.githubusercontent.com/42647570/151579317-ed1ffb64-f76b-44d7-b80b-f858d45e2e46.png)

```bash
zypper lu --all
Loading repository data...
Reading installed packages...
S | Repository          | Name                 | Current Version      | Available Version    | Arch
--+---------------------+----------------------+----------------------+----------------------+-------
v | Uyuni Server Stable | Uyuni-Server-release | 2021.09-141.3.uyuni1 | 2021.12-160.4.uyuni1 | x86_64

# zypper up --allow-vendor-change
Loading repository data...
Reading installed packages...

The following package is going to be upgraded:
  Uyuni-Server-release

The following product is going to be upgraded:
  Uyuni Server  2021.09-0 -> 2021.12-0

The following product is going to change vendor:
  Uyuni Server  openSUSE -> obs://build.opensuse.org/systemsmanagement:Uyuni
```

If more information is needed I will gladly provide it!
Thank you!

**Comment by lumarel** [comment]
Another short update, also from 2022.02 to 2022.03 the same Uyuni-Server-release package had the sam Vendor change again 🙂

**Comment by juliogonzalez** [comment]
Seems this could be a problem with the product builder.

Here are the conclusions I sent to Adrian, after talking to the zypp team:

```
Thanks to Benjamin and Michael Andres, we found something interesting.

The vendor defined at the .prod file inside the release package doesn't match 
the vendor for the release package itself.

For example, take the latest release package for Uyuni 2022.05:

https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable/
images/repo/Uyuni-Server-POOL-x86_64-Media1/x86_64/Uyuni-Server-
release-2022.05-180.1.uyuni1.x86_64.rpm

$ rpm -qi Uyuni-Server-release-2022.05-180.1.uyuni1.x86_64.rpm
Name        : Uyuni-Server-release
Version     : 2022.05
Release     : 180.1.uyuni1
Architecture: x86_64
Install Date: (not installed)
Group       : System/Fhs
Size        : 1455
License     : GPL-2.0-or-later
Signature   : RSA/SHA256, Tue May  3 15:59:15 2022, Key ID 972e5d6c0d20833e
Source RPM  : Uyuni-Server-release-2022.05-180.1.uyuni1.src.rpm
Build Date  : Tue May  3 15:59:12 2022
Build Host  : lamb22
Relocations : (not relocatable)
Vendor      : obs://build.opensuse.org/systemsmanagement:Uyuni
Summary     : Uyuni Server
Description :
Uyuni lets you efficiently manage physical, virtual,
and cloud-based Linux systems. It provides automated and cost-effective
configuration and software management, asset management, and system
provisioning.
Distribution: systemsmanagement:Uyuni:Master / openSUSE_Leap_15.3

But if you extract the contents and check etc/products.d/Uyuni-Server.prod
you will see:

  <vendor>openSUSE</vendor>

Could it be something changed at the product builder lately?

**Comment by mbussolotto** [comment]
@lumarel please consider that the current upgrade (to 2022.06) is a major, so you should follow this https://www.uyuni-project.org/uyuni-docs/en/uyuni/installation-and-upgrade/server-major-upgrade-uyuni.html ( and major upgrade can have vendor changes, since `server-migrator.sh` script contains `zypper -n dup --allow-vendor-change` ). Please provide us any feedback after the upgrade, thanks!

**Comment by lumarel** [comment]
Hi @mbussolotto,

Sorry for the late reply, didn't have the time to do the upgrade earlier,
anyway, these are the vendor changes during the `server-migrator.sh` run:

```
The following 11 packages are going to change vendor:
  c3p0                                           obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  golang-github-QubitProducts-exporter_exporter  obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  httpcomponents-asyncclient                     obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  jsch                                           obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  mchange-commons                                obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  prometheus-postgres_exporter                   obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  python3-dmidecode                              obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  python3-hwdata                                 obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  python3-jabberpy                               obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  python3-urlgrabber                             obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>
  supportutils-plugin-salt                       obs://build.opensuse.org/systemsmanagement:Uyuni -> SUSE LLC <https://www.suse.com/>

The following product is going to change vendor:
  Uyuni Server  openSUSE -> obs://build.opensuse.org/systemsmanagement:Uyuni
```

So as this was a major upgrade I will do the upgrade with the debug-resolver on the next minor release 👍🏻

**Comment by lumarel** [comment]
> next time you see this, please run the following before doing the actual update

@juliogonzalez I created a tarball with the solver data now, still happening.

```
The following 4 package updates will NOT be installed:
  Uyuni-Server-release cobbler netty supportutils-plugin-salt

The following product update will NOT be installed:
  "Uyuni Server"
```

[zypper.solverTestCase.tar.gz](https://github.com/uyuni-project/uyuni/files/9316955/zypper.solverTestCase.tar.gz)

**Comment by juliogonzalez** [comment]
> > next time you see this, please run the following before doing the actual update
> 
> @juliogonzalez I created a tarball with the solver data now, still happening.
> 
> ```
> The following 4 package updates will NOT be installed:
>   Uyuni-Server-release cobbler netty supportutils-plugin-salt
> 
> The following product update will NOT be installed:
>   "Uyuni Server"
> ```
> 
> [zypper.solverTestCase.tar.gz](https://github.com/uyuni-project/uyuni/files/9316955/zypper.solverTestCase.tar.gz)

`openSUSE -> obs://build.opensuse.org/systemsmanagement:Uyuni` or the other way around this time?

**Comment by lumarel** [comment]
I'm not completely sure anymore unfortunately, but if I remember correctly, everything beside Uyuni-Server-release and "Uyuni Server" changed from uyuni to openSUSE and these two to uyuni

**Comment by lumarel** [comment]
It looks like this issue got fixed with 2022.12!
Thank you for further looking into it!

**Comment by juliogonzalez** [comment]
If this the case, looks to me something at OBS changed again to get this fixed, as I didn't get a reply about the possible causes from the OBS specialists.

Anyway, happy to see the problem is gone now :-)
